src/daemon/daemon.cpp

```diff
@@ -60,19 +60,27 @@ struct t_internals {
 public:
   t_core core;
   t_p2p p2p;
-  t_rpc rpc;
+  std::vector<std::unique_ptr<t_rpc>> rpcs;
 
   t_internals(
       boost::program_options::variables_map const & vm
     )
     : core{vm}
     , protocol{vm, core}
     , p2p{vm, protocol}
-    , rpc{vm, core, p2p}
   {
     // Handle circular dependencies
     protocol.set_p2p_endpoint(p2p.get());
     core.set_protocol(protocol.get());
+
+    const auto testnet = command_line::get_arg(vm, cryptonote::arg_testnet_on);
+    const auto restricted = command_line::get_arg(vm, cryptonote::core_rpc_server::arg_restricted_rpc);
+    const auto main_rpc_port = command_line::get_arg(vm, testnet ? cryptonote::core_rpc_server::arg_testnet_rpc_bind_port : cryptonote::core_rpc_server::arg_rpc_bind_port);
+    const auto restricted_rpc_port = command_line::get_arg(vm, testnet ? cryptonote::core_rpc_server::arg_testnet_rpc_restricted_bind_port : cryptonote::core_rpc_server::arg_rpc_restricted_bind_port);
+    rpcs.emplace_back(new t_rpc{vm, core, p2p, restricted, testnet, main_rpc_port, "core"});
+
+    if(restricted_rpc_port.size())
```

Comment:
That should probably be somhting like "if (command_line::is_arg_defaulted(vm, ...))" and have the get_arg inside the if.